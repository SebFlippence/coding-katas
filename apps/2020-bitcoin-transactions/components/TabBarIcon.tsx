import * as React from 'react';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import colors from '../constants/colors';

export default function TabBarIcon({ name, focused }) {
  return (
    <MaterialCommunityIcons
      name={name}
      size={30}
      style={{ marginBottom: -3 }}
      color={focused ? colors.tabIconSelected : colors.tabIconDefault}
    />
  );
}
