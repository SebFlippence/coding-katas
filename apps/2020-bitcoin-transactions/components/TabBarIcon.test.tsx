import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import TabBarIcon from './TabBarIcon';


describe('TabBarIcon', () => {
  const renderer = new ShallowRenderer();

  test('rendering in focus', () => {
    const tree = renderer.render(<TabBarIcon name="access-point" focused />);
    expect(tree).toMatchSnapshot();
  });
  
  test('rendering out of focus', () => {
    const tree = renderer.render(<TabBarIcon name="access-point" focused={false} />);
    expect(tree).toMatchSnapshot();
  });
});
