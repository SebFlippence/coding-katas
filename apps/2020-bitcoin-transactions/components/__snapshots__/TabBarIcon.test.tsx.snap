// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon rendering in focus 1`] = `
<Icon
  allowFontScaling={false}
  color="#2f95dc"
  name="access-point"
  size={30}
  style={
    Object {
      "marginBottom": -3,
    }
  }
/>
`;

exports[`TabBarIcon rendering out of focus 1`] = `
<Icon
  allowFontScaling={false}
  color="#ccc"
  name="access-point"
  size={30}
  style={
    Object {
      "marginBottom": -3,
    }
  }
/>
`;
